ruby/ext/google/protobuf_c/message.c

```diff
@@ -70,6 +70,36 @@ VALUE Message_alloc(VALUE klass) {
   return ret;
 }
 
+static VALUE which_oneof_field(MessageHeader* self, const upb_oneofdef* o) {
+  // If no fields in the oneof, always nil.
+  if (upb_oneofdef_numfields(o) == 0) {
```

Comment:
Define a constant for 0 (and use it everywhere 0 is being used for this purpose). Comment that 0 is safe because it's an invalid field number.